fastfield_codecs/src/linearinterpol.rs

```diff
@@ -91,17 +91,26 @@ impl FastFieldCodecReader for LinearInterpolFastFieldReader {
 pub struct LinearInterpolFastFieldSerializer {}
 
 #[inline]
-fn get_slope(first_val: u64, last_val: u64, num_vals: u64) -> f32 {
+pub(crate) fn get_slope(first_val: u64, last_val: u64, num_vals: u64) -> f32 {
     if num_vals <= 1 {
         return 0.0;
     }
     //  We calculate the slope with f64 high precision and use the result in lower precision f32
     //  This is done in order to handle estimations for very large values like i64::MAX
-    ((last_val as f64 - first_val as f64) / (num_vals as u64 - 1) as f64) as f32
+    let diff = diff(last_val, first_val);
+    (diff / (num_vals - 1) as f64) as f32
+}
+
+fn diff(val1: u64, val2: u64) -> f64 {
```

Comment:
Can you add a comment?

For which values is this dance useful?

Reply:
I added a comment